plugins/lua/sort.lua

```diff
@@ -125,6 +125,109 @@ local function get_max_skill(unit_id, list)
     return max
 end
 
+local function melee_skill_effectiveness(unit_id, skill_list)
+    local unit = df.unit.find(unit_id)
+
+    -- Physical attributes
+    local strength = dfhack.units.getPhysicalAttrValue(unit, df.physical_attribute_type.STRENGTH)
+    local agility = dfhack.units.getPhysicalAttrValue(unit, df.physical_attribute_type.AGILITY)
+    local toughness = dfhack.units.getPhysicalAttrValue(unit, df.physical_attribute_type.TOUGHNESS)
+    local endurance = dfhack.units.getPhysicalAttrValue(unit, df.physical_attribute_type.ENDURANCE)
+    local body_size_base = unit.body.size_info.size_base
+
+    -- Mental attributes
+    local willpower = dfhack.units.getMentalAttrValue(unit, df.mental_attribute_type.WILLPOWER)
+    local spatial_sense = dfhack.units.getMentalAttrValue(unit, df.mental_attribute_type.SPATIAL_SENSE)
+    local kinesthetic_sense = dfhack.units.getMentalAttrValue(unit, df.mental_attribute_type.KINESTHETIC_SENSE)
+
+    -- Skills
+    local melee_skill = get_max_skill(unit_id, skill_list)
+    if melee_skill then
+        melee_skill = melee_skill.rating
+    else
+        melee_skill = 0
+    end
```

Comment:
this confuses some IDEs since `melee_skill` appears to change types. it's better to use different vars for this.